### PR TITLE
cli: print a message about new and resolved conflicts

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Review by PhilipMetzger — APPROVED** [review]
LGTM, it's only a missing a Fixes #555 in the commit message.

**Comment by martinvonz** [comment]
> LGTM, it's only a missing a Fixes #555 in the commit message.

That bug also mentioned divergent changes and conflicted branches. Perhaps it should have been 3 separate bugs. Maybe I'll close it and open new ones for the remaining bits once this is in.

File: cli/src/cli_util.rs [hunk 1]
```diff
@@ -1583,7 +1583,64 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
...
+                    )?;
+                }
+                for commit in root_conflict_commits {
+                    writeln!(fmt, "  jj new {}", short_change_hash(commit.change_id()))?;
+                }
+                writeln!(
+                    fmt,
+                    "Then use `jj resolve`, or edit the conflict markers in the file directly.
+Once the conflicts are resolved, you may want inspect the result with `jj diff`.
+Then run `jj squash` to move the resolution into the conflicted commit.",
```
Comment (PhilipMetzger) [inline cli/src/cli_util.rs:1638]: Should probably be a raw string.

**Comment by PhilipMetzger** [comment]
> That bug also mentioned divergent changes and conflicted branches. Perhaps it should have been 3 separate bugs. Maybe I'll close it and open new ones for the remaining bits once this is in.

SGTM, as long as we close them when this lands.

File: cli/src/cli_util.rs (hunk 1 above)
Comment (martinvonz) [reply cli/src/cli_util.rs:1638]: Can you remind me why that's better?

Comment (PhilipMetzger) [reply cli/src/cli_util.rs:1638]: Mostly preventing mistakes when the text changes, as manually inserting `\n` and `\` to continue a text block is error-prone.

Comment (martinvonz) [reply cli/src/cli_util.rs]: Ah, that's reasonable. I have generally used raw strings only when necessary.

**Comment by martinvonz** [comment]
I'll leave this until after the release on Wednesday since it's pretty large UX change and might have unknown rough edges.

**Comment by martinvonz** [comment]
I'll probably merge this in a day or two if there are no further reviews

**Pull request merged**